utils/python/CIME/build.py

```diff
@@ -274,25 +285,41 @@ def case_build(caseroot, testmode):
     # Load modules
     env_module = EnvModule(mach, compiler, cimeroot, caseroot, mpilib, debug)
     env_module.load_env_for_case()
-    # Need to flush case xml to disk before calling preview_namelists
-    case.flush()
-    run_cmd("./preview_namelists")
 
-    build_checks(case, build_threaded, comp_interface, use_esmf_lib, mpilib,
+    if not sharedlib_only:
+        # Need to flush case xml to disk before calling preview_namelists
+        case.flush()
```

Comment:
I think that this case.flush() needs to happen in both passes - the case may have been updated with something needed for the sharedlib build.